internal/terraform/terraform.go

```diff
@@ -333,3 +289,71 @@ func OutputSingleValue(dryRun bool, directory, tfEntrypoint, outputName string)
 
 	log.Println("tfOutput is: ", tfOutput.String())
 }
+
+// ApplyUsersTerraform load environment variables into the host based on the git provider, change directory to the
+// Terraform required modules, terraform init, terraform apply and clean terraform files.
+// todo: break it into smaller functions with no dependencies in order to allow unit tests
+func ApplyUsersTerraform(dryRun bool, directory string, gitProvider string) error {
+
+	config := configs.ReadConfig()
+
+	if viper.GetBool("create.terraformapplied.users") || dryRun {
+		log.Println("skipping: ApplyUsersTerraform")
+		return nil
+	}
+
+	if len(gitProvider) == 0 {
+		return errors.New("git provider not provided, skipping terraform apply")
+	}
+
+	log.Println("Executing ApplyUsersTerraform")
+
+	//* AWS_SDK_LOAD_CONFIG=1
+	//* https://registry.terraform.io/providers/hashicorp/aws/2.34.0/docs#shared-credentials-file
+	envs := map[string]string{}
+
+	if gitProvider == "github" {
```

Comment:
I believe it would be better to rely on gitlab domain to check if gitlab is available or not. if we remove github dependency some day, gitlab will be wired to a github value.